src/elements/element.line.js

```diff
@@ -43,17 +40,17 @@ module.exports = Element.extend({
 		ctx.save();
 
 		// Stroke Line Options
-		ctx.lineCap = vm.borderCapStyle || globalOptionLineElements.borderCapStyle;
+		ctx.lineCap = valueOrDefault(vm.borderCapStyle, globalOptionLineElements.borderCapStyle);
```

Comment:
I think the fallback is mostly for the case where an invalid value is set. I'd be a bit wary about remove the fallback here since it has existed for a while

Reply:
> I think the fallback is mostly for the case where an invalid value is set

I think elements should not have to deal with the options fallback, only controllers should implement that logic. If for any reason the element model contains `undefined` values, it should be considered a bug or it's expected and the element knows how to handle this undefined value (e.g. `lineDashOffset === undefined` could mean no dash). 

We should maybe try to remove these defaults fallback from elements since it silents bugs at the controller level?

Or is it really too risky?

Reply:
This might be a good reason to close this PR without merging it. Instead we'd do it at the controller level. However that would mean that changing the `defaultColor` would require calling `update()` afterwards. I think that's acceptable to me

Reply:
I think we still need to remove the line below (and similar lines) otherwise the changes in global defualt options won’t be applied when `update` is called.
https://github.com/chartjs/Chart.js/blob/e07c5d0c6c39e621fbe79de39d6e2de66da1cd9a/src/elements/element.line.js#L15

Reply:
I agree with @simonbrunel, we should remove this fallback from elements.
I also agree with @etimberg in that its ok to require `update` after change in defaults.

Considering that all internal controllers have new fallback logic after 2.7.3, fallback removal in elements should make no difference.

So I'd either remove the fallback or revert to using `||` if removal seems too risky.

Reply:
I removed the fallbacks and did this at entire change at the resolve level